addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/AbstractNetatmoClimateThingHandler.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.handler;
+
+import static org.openhab.binding.netatmo.NetatmoBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.netatmo.config.AbstractNetatmoClimateThingConfiguration;
+import org.openhab.binding.netatmo.internal.ChannelTypeUtils;
+
+import io.swagger.client.model.NAMeasureResponse;
+
+/**
+ * {@link AbstractNetatmoClimateThingHandler} is the abstract class that handles
+ * common behaviors of both Devices and Modules
+ *
+ * @author Gaël L'hopital - Initial contribution OH2 version
+ * @author Ing. Peter Weiss
+ */
+abstract class AbstractNetatmoClimateThingHandler<X extends AbstractNetatmoClimateThingConfiguration>
+        extends AbstractNetatmoThingHandler<X> {
+    private final List<Integer> signalThresholds = new ArrayList<Integer>();
+    protected List<String> measuredChannels = new ArrayList<String>();
+    protected NAMeasureResponse measures = null;
+
+    AbstractNetatmoClimateThingHandler(Thing thing, Class<X> configurationClass) {
+        super(thing, configurationClass);
+    }
+
+    @Override
+    public void initialize() {
+        String signalLevels = getProperty(PROPERTY_SIGNAL_LEVELS);
```

Comment:
I believe that @clinique changed that to put the loading of porperty in getSignalStrength()